src/Tickets/QR/Controller.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace TEC\Tickets\QR;
+
+use \TEC\Common\Contracts\Provider\Controller as Controller_Contract;
+
+/**
+ * Class Controller.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Tickets\QR
+ */
+class Controller extends Controller_Contract {
+	/**
+	 * Determines if this controller will register.
+	 * This is present due to how UOPZ works, it will fail if method belongs to the parent/abstract class.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function is_active(): bool {
+		return true;
+	}
+
+	/**
+	 * Boot the Controller.
+	 *
+	 * This function is used to instantiate the singleton classes and register any other providers.
+	 *
+	 * @since   TBD
+	 */
+	public function boot() {
+		$this->container->bind( Facade::class, [ $this, 'bind_facade_or_error' ] );
```

Comment:
Do not use the `boot` method to register a binding, use the `do_register` method of the Controller.
The `boot` method is meant to contain methods that would **build** things that are required at the application boot time, in contrast with the application initialization phase that will rely on the Providers/Controllers `do_register` methods.